exporter/elasticsearchexporter/integrationtest/datareceiver.go

```diff
@@ -40,22 +54,30 @@ func newElasticsearchDataReceiver(t testing.TB) *esDataReceiver {
 	}
 }
 
-func (es *esDataReceiver) Start(_ consumer.Traces, _ consumer.Metrics, lc consumer.Logs) error {
+func (es *esDataReceiver) Start(tc consumer.Traces, _ consumer.Metrics, lc consumer.Logs) error {
 	factory := receiver.NewFactory(
 		component.MustNewType("mockelasticsearch"),
 		createDefaultConfig,
 		receiver.WithLogs(createLogsReceiver, component.StabilityLevelDevelopment),
+		receiver.WithTraces(createTracesReceiver, component.StabilityLevelDevelopment),
 	)
```

Comment:
Can we avoid the global var by storing the `receivers` in a closure? e.g.
```suggestion
        receivers := sharedcomponent.NewSharedComponents()
	factory := receiver.NewFactory(
		component.MustNewType("mockelasticsearch"),
		createDefaultConfig,
		receiver.WithLogs(createLogsReceiverFunc(receivers), component.StabilityLevelDevelopment),
		receiver.WithTraces(createTracesReceiverFunc(receivers), component.StabilityLevelDevelopment),
	)
```

Reply:
If multiple call to `Start` is made we will create multiple copies of the receiver which is not expected. I think this is the common way to use shared component.